### PR TITLE
[Python] Add Portuguese support for DateTimeV2

### PR DESCRIPTION
**Pull request by samhickey25**
Add support for datetime entities in python for Portuguese and update the Specs to reflect these changes

### CODE REVIEW

**Review by tellarin — CHANGES_REQUESTED** [review]
Build is failing. Can you take a look? We can review once build passes.

**Comment by tellarin** [comment]
@samhickey25, BTW, I noticed that in the fork where this PR come from you folks seem to have Python support for Italian Numbers. Are you also looking at contributing that back? That would be very helpful. As well as any other contributions. :-)

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/portuguese/parsers.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            result.future_value = result.past_value
+
+            result.success = True
+            return result
+
+        start_index = len(PortugueseDateTime.Tomorrow) if trimmed_text.startswith(
+            PortugueseDateTime.Tomorrow) else 0
+
+        # handle Date followed by morning, afternoon
+        # Add handling code to handle morning, afternoon followed by Date
```
Comment (tellarin) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/portuguese/parsers.py:56]: Is this replicating the state of the .NET code or a Python-specific workaround?

File: Specs/DateTime/Portuguese/DateExtractor.json [hunk 2]
```diff
@@ -695,7 +695,7 @@
   },
   {
     "Input": "Você está livre em 13.5.2015",
-    "NotSupported": "python",
+    "NotSupported": "",
```
Comment (tellarin) [inline Specs/DateTime/Portuguese/DateExtractor.json:698]: Such empty NotSupported tags can be removed to keep specs clean.

File: Specs/DateTime/Portuguese/DateParser.json [hunk 3]
```diff
@@ -1396,7 +1396,7 @@
     "Context": {
       "ReferenceDateTime": "2016-11-07T00:00:00"
     },
-    "NotSupported": "python",
+    "NotSupported": "",
```
Comment (tellarin) [inline Specs/DateTime/Portuguese/DateParser.json:1399]: Same.

File: Specs/DateTime/Portuguese/TimeParser.json [hunk 4]
```diff
@@ -883,7 +883,7 @@
   },
   {
     "Input": "Voltarei às 7h01",
-    "NotSupported": "python",
+    "NotSupported": "",
```
Comment (tellarin) [inline Specs/DateTime/Portuguese/TimeParser.json:886]: Same.

**Review by tellarin — CHANGES_REQUESTED** [review]
Minor changes to make.
I also noticed harcoded strings that should be refactored in YAML (along with TODOs). i assume these come as-is from .NET? I'm OK with merging this version.

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/portuguese/parsers.py (hunk 1 above)
Comment (samhickey25) [reply Python/libraries/recognizers-date-time/recognizers_date_time/date_time/portuguese/parsers.py:56]: This is replicating .NET behaviour

**Comment by samhickey25** [comment]
> Minor changes to make. I also noticed harcoded strings that should be refactored in YAML (along with TODOs). i assume these come as-is from .NET? I'm OK with merging this version.

Yeah these files follow the .NET files and behaviour closely

**Review by tellarin — APPROVED** [review]

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/portuguese/parsers.py [hunk 5]
```diff
@@ -54,7 +54,7 @@ def parse_specific_time_of_day(self, source: str, reference: datetime) -> DateTi
 
         # handle Date followed by morning, afternoon
         # Add handling code to handle morning, afternoon followed by Date
-        # Add handling code to handle early/late morning, afternoon
+        # Add handling code to handle early/late morning, afternoon.
```
Comment (tellarin) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/portuguese/parsers.py:57]: No need for such commits. We can either re-run the same build. Or close/reopen the PR and a new build is triggered.

**Pull request merged**